extensions/amp-nexxtv-player/0.1/amp-nexxtv-player.js

```diff
@@ -134,25 +161,55 @@ class AmpNexxtvPlayer extends AMP.BaseElement {
       src += `&streamingFilter=${encodeURIComponent(streamingFilter)}`;
     }
 
-    return assertAbsoluteHttpOrHttpsUrl(src);
-  }
+    if (exitMode !== null && exitMode.length > 0) {
+      src += `&exitMode=${encodeURIComponent(exitMode)}`;
+    }
 
-  /** @override */
-  layoutCallback() {
-    const iframe = createFrameFor(this, this.getVideoIframeSrc_());
+    if (this.consentString_ !== null){
+      src += `&consentString=${encodeURIComponent(this.consentString_)}`;
+    }
 
-    this.iframe_ = iframe;
+    return assertAbsoluteHttpOrHttpsUrl(src);
+  }
 
-    this.unlistenMessage_ = listen(this.win, 'message', (event) => {
-      this.handleNexxMessage_(event);
+  /**
+   * Get consent data from consent module
+   * @return {Promise[]}
+   */
+  getConsents_() {
+    const consentPolicy = super.getConsentPolicy() || 'default';
+    const consentPromise = consentPolicy
+      ? getConsentPolicyState(this.element, consentPolicy)
+      : Promise.resolve(null);
+    const sharedDataPromise = consentPolicy
+      ? getConsentPolicySharedData(this.element, consentPolicy)
+      : Promise.resolve(null);
+    const consentStringPromise = consentPolicy
+      ? getConsentPolicyInfo(this.element, consentPolicy)
+      : Promise.resolve(null);
+
+    return Promise.all([
+      consentPromise,
+      sharedDataPromise,
+      consentStringPromise,
+    ]).then((consentData) => {
+      this.consentState_ = consentData[0];
+      this.consentSharedData_ = consentData[1];
+      this.consentString_ = consentData[2];
     });
+  }
 
-    this.element.appendChild(this.iframe_);
-    const loaded = this.loadPromise(this.iframe_).then(() => {
-      dispatchCustomEvent(this.element, VideoEvents.LOAD);
+  /** @override */
+  layoutCallback() {
+    return this.getConsents_().then(() => {
+      this.iframe_ = createFrameFor(this, this.getVideoIframeSrc_());
+      this.unlistenMessage_ = listen(this.win, 'message', (event) => {
+        this.handleNexxMessage_(event);
+      });
+      return this.loadPromise(this.iframe_).then(
+        () => this.playerReadyPromise_
+      );
```

Comment:
You should probably add a test confirming that the consent value is passed through.

Reply:
I have added a test, but could not run the test locally since all tests drop Error: Timeout of 2000ms exceeded. I'm not sure how to fix this.